interfaces/cython/cantera/composite.py

```diff
@@ -516,21 +516,34 @@ def __init__(self, phase, shape=(0,), states=None, extra=None):
         reserved = self.__dir__()
 
         self._extra = {}
-        if isinstance(extra, dict):
+
+        if isinstance(extra,str):
+            extra = [extra]
+
+        if isinstance(extra, list) and all(isinstance(name, str) for name in extra):
```

Comment:
extra could also be an instance of `tuple` or `ndarray` of strings? I believe all of these are supported for when no shape is used for the instantiation of `SolutionArray` (i.e. I am talking about the case `elif extra and self._shape == (0,):`).

Reply:
that'd be a quick fix but I don't think np.arrays are important?. 
Besides I believe the other part should be removed where `self._shape` is trivial as after this commit it would only be a desperate attempt to map something in extra, probably not useful (is there a particular use case?). it lacks checks and something weird like integers, for instance can pass which can't be used back.e.g.

`>>>>states3 = ct.SolutionArray(gas, shape=(0,0,0), extra=[1])`

Reply:
I don’t think that the alternative case can be executed for useful content either (you already caught all useful cases), I.e. I believe it can be deleted. 

On a formatting note, there are a couple of spaces missing after commas (e.g. it should be `shape=(0, 0, 0)` in the previous post; same for some other lines you edited). It’s minor, but the code should remain in compliance with PEP 8.

Reply:
Yes the unit tests pass.
sorry for the formatting though.